src/test/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcherTest.java

```diff
@@ -79,4 +92,349 @@ public void testShouldUpdate() throws IOException {
         repository);
     assertFalse(fetcher.shouldUpdate(freshRepository));
   }
+
+  @Test
+  public void testUpdateGithubInstance() throws IOException {
+    GitHub github = mock(GitHub.class);
+    fetcher = spy(new TestGitHubDataFetcher(github));
+    assertEquals(fetcher.github(), github);
+
+    // Create the second GitHub mock bean
+    GitHub github2 = mock(GitHub.class);
+    fetcher = spy(new TestGitHubDataFetcher(github2));
+    assertEquals(fetcher.github(), github2);
+  }
+
+  @Test
+  public void testLoadAndCleanRepository() throws IOException, InterruptedException {
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testfile");
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      localRepositoryFor(project, 1);
+
+      cleanup(project, 0);
+    }
+  }
+
+  @Test
+  public void testCleanAndLoadRepository() throws IOException, InterruptedException {
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testFile");
+
+      cleanup(project, 0);
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      localRepositoryFor(project, 1);
+    }
+  }
+
+  @Test
+  public void testMultiGithubInstanceAndAccessRepo() throws IOException {
+    // Create the first project with first GitHub mock instance.
+    GitHubProject firstProject = new GitHubProject("test", "project1");
+    Path project1Dir = projectDir(firstProject.name());
+    GitHub github = mock(GitHub.class);
+    try (Repository repository =
+        FileRepositoryBuilder.create(project1Dir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, firstProject, "testFile1");
+
+
+      fetcher = spy(new TestGitHubDataFetcher(github));
+
+      fetcher.addRepositoryInfoForTesting(firstProject, project1Dir);
+
+      localRepositoryFor(firstProject, 1);
+      assertEquals(fetcher.github(), github);
```

Comment:
I am not sure if I understand the purpose of this test. First, it creates a fetcher with an instance of `GitHub`. Then, it loads a local repository. Finally, it checks if `github()` returns an expected instance of `GitHub`. But loading local repositories doesn't use `GitHub`. The `testUpdateGithubInstance()` already checks that setting an instance of `GitHub` via constrictor works fine.

Reply:
You are right. This test actually doesn't test anything new. Earlier I had written the test case with the mindset that github plays a crucial role in gathering information about local repositories. But we are not testing that part.